### ISSUE TITLE
auxiliary loss

### ISSUE THREAD

**Issue by wwjwy** [open-mmlab/mmsegmentation#802]
When using FCNHead as the auxiliary header, does it mean that the output of a certain middle layer (in_index=2) of the backbone network (such as resnet101) is up-sampled to the input size (512x512), and then the auxiliary loss is calculated?

**Comment by Junjun2016** [comment]
Hi @wwjwy 
Exactly!
mmseg\models\decode_heads\decode_head.py L213-233.

**Comment by wwjwy** [comment]
I noticed that the auxiliary loss and decoder loss weights are 0.4 and 1.0 respectively, so the total loss=0.4*auxiliary loss+1.0*decoder loss？

**Comment by MengzhangLI** [comment]
Hi, @wwjwy

You are right, each auxiliary head would upsample its feature map to input size to calculate loss together.

For example, in cityscapes dataset and input is `(1024, 1024)`certain feature map size is `(4, 19, 256, 256)` where `4` is batch size and `19` is class number. It would be upsampled to `(4, 19, 1024, 1024)` whose size is same of input.

Furthermore, auxiliary head and its loss only join the training mode, they do not participate in the inference/test mode.

Best,

**Comment by wwjwy** [comment]
> Hi, @wwjwy
> 
> You are right, each auxiliary head would upsample its feature map to input size to calculate loss together.
> 
> For example, in cityscapes dataset and input is `(1024, 1024)`certain feature map size is `(4, 19, 256, 256)` where `4` is batch size and `19` is class number. It would be upsampled to `(4, 19, 1024, 1024)` whose size is same of input.
> 
> Furthermore, auxiliary head and its loss only join the training mode, they do not participate in the inference/test mode.
> 
> Best,

Thanks a lot

**Comment by wwjwy** [comment]
> Hi, @wwjwy
> 
> You are right, each auxiliary head would upsample its feature map to input size to calculate loss together.
> 
> For example, in cityscapes dataset and input is `(1024, 1024)`certain feature map size is `(4, 19, 256, 256)` where `4` is batch size and `19` is class number. It would be upsampled to `(4, 19, 1024, 1024)` whose size is same of input.
> 
> Furthermore, auxiliary head and its loss only join the training mode, they do not participate in the inference/test mode.
> 
> Best,

I noticed that the auxiliary loss and decoder loss weights are 0.4 and 1.0 respectively, so the total loss=0.4auxiliary loss+1.0decoder loss？
But my model output loss = auxiliary loss + decoder loss, without considering the loss weight

**Comment by wwjwy** [comment]
for example  decode.loss_seg:0.0083,  aux.loss_seg:0.0051,  loss:0.0135

**Comment by MengzhangLI** [comment]
If I remember well, the weight `0.4` should be used on auxiliary loss, please check the code agian.

Best,

**Comment by wwjwy** [comment]
> If I remember well, the weight `0.4` should be used on auxiliary loss, please check the code agian.
> 
> Best,

yes, you are right.  but how the total loss was calculated? should not 0.4*auxiliary loss+1.0*decoder_loss=total loss?

**Comment by MengzhangLI** [comment]
> > If I remember well, the weight `0.4` should be used on auxiliary loss, please check the code agian.
> > Best,
> 
> yes, you are right. but how the total loss was calculated? should not 0.4_auxiliary loss+1.0_decoder_loss=total loss?

I mean the `aux.loss_seg` the log shows has been multiplied `0.4`. Its original auxiliary loss is `0.0051 / 0.4 = 0.01275`. Please check whether I am right.

Best,

**Comment by wwjwy** [comment]
> > > If I remember well, the weight `0.4` should be used on auxiliary loss, please check the code agian.
> > > Best,
> > 
> > 
> > yes, you are right. but how the total loss was calculated? should not 0.4_auxiliary loss+1.0_decoder_loss=total loss?
> 
> I mean the `aux.loss_seg` the log shows has been multiplied `0.4`. Its original auxiliary loss is `0.0051 / 0.4 = 0.01275`. Please check whether I am right.
> 
> Best,

i got it, thank you very much!
